 Fix the NullPointerException
Checking Account object for NULL before accessing it.
 Fix to NullPointerException when activity is relaunched
When ChooseTypeAndAccountActivity is relaunched and resumed by ActivityManager a call
is made to onActivityResult without first triggering onResume.
This leads to mAccounts object still being null
Fixed by properly setting mAccounts object if it is null prior to use after function
onCreate has been called
 [ActivityManager] Reset killed flag at attachApplicationLocked
A restarted process will reuse original process record.
If only reset KilledByAm flag, not reset Killed flag.
When app process have been killed by AMS can skip  unnessary ANR,
but not catch lowmemorykiller in AppErrors.appNotResponding()

diff --git a/core/java/android/accounts/ChooseTypeAndAccountActivity.java b/core/java/android/accounts/ChooseTypeAndAccountActivity.java
index 0262f54..f9b5deb 100644
--- a/core/java/android/accounts/ChooseTypeAndAccountActivity.java
+++ b/core/java/android/accounts/ChooseTypeAndAccountActivity.java
@@ -293,7 +293,7 @@ protected void onActivityResult(final int requestCode, final int resultCode,
         if (resultCode == RESULT_CANCELED) {
             // if canceling out of addAccount and the original state caused us to skip this,
             // finish this activity
-            if (mAccounts.isEmpty()) {
+            if (null == mAccounts || mAccounts.isEmpty()) {
                 setResult(Activity.RESULT_CANCELED);
                 finish();
             }
diff --git a/core/java/android/accounts/ChooseTypeAndAccountActivity.java b/core/java/android/accounts/ChooseTypeAndAccountActivity.java
index 0062f54..f9b5deb 100644
--- a/core/java/android/accounts/ChooseTypeAndAccountActivity.java
+++ b/core/java/android/accounts/ChooseTypeAndAccountActivity.java
@@ -250,6 +250,10 @@ protected void onSaveInstanceState(final Bundle outState) {
             outState.putParcelableArray(KEY_INSTANCE_STATE_EXISTING_ACCOUNTS, mExistingAccounts);
         }
         if (mSelectedItemIndex != SELECTED_ITEM_NONE) {
+            if (mAccounts == null) {
+                final AccountManager accountManager = AccountManager.get(this);
+                mAccounts = getAcceptableAccountChoices(accountManager);
+            }
             if (mSelectedItemIndex == mAccounts.size()) {
                 outState.putBoolean(KEY_INSTANCE_STATE_SELECTED_ADD_ACCOUNT, true);
             } else {
@@ -291,6 +295,10 @@ protected void onActivityResult(final int requestCode, final int resultCode,
         mPendingRequest = REQUEST_NULL;
 
         if (resultCode == RESULT_CANCELED) {
+            if (mAccounts == null) {
+                final AccountManager accountManager = AccountManager.get(this);
+                mAccounts = getAcceptableAccountChoices(accountManager);
+            }
             // if canceling out of addAccount and the original state caused us to skip this,
             // finish this activity
             if (null == mAccounts || mAccounts.isEmpty()) {
